src/classes/CDL_CascadeDeleteLookups.cls

```diff
@@ -0,0 +1,349 @@
+                    result.add(errorType + ' (' + getPlainTextChildrenList(childrenByErrorType.get(errorType)) + ')');
+                }
+                return String.format(Label.CascadeDeletionErrorLightning, new List<String>{deletedRecordName, String.join(result, ',')});
+            }
+            else {
+                for (String errorType : childrenByErrorType.keySet()) {
+                    result.add(errorType + ':' + getHTMLChildrenList(childrenByErrorType.get(errorType)));
+                }
+                return String.format(Label.CascadeDeletionError, new List<String>{deletedRecordName, String.join(result, '<br/>')});
+            }
+        }
+
+        /*******************************************************************************************************
+        * @description Returns a HTML list of the child records preventing the deletion, with links to those records.
+        * @param Map<Id, String> The child records preventing the deletion.
+        * @return String The HTML text that contains a list of the child records.
+        ********************************************************************************************************/
+        private String getHTMLChildrenList(Map<Id, String> childRecords) {
+            List<String> result = new List<String>{'<ul>'};
+            for (Id childrenRecordId : childRecords.keySet()) {
+                result.add(String.format(HTML_TEXT_CHILD_RECORDS_FORMAT, new List<String>{childrenRecordId, childRecords.get(childrenRecordId)}));
+            }
+            result.add('</ul>');
+            
+            return String.join(result, '');
+        }
+
+        /*******************************************************************************************************
+        * @description Returns a plain text list of the child records preventing the deletion.
+        * @param Map<Id, String> The child records preventing the deletion.
+        * @return String The plain text that contains a list of the child records.
+        ********************************************************************************************************/
+        private String getPlainTextChildrenList(Map<Id, String> childRecords) {
+            List<String> result = new List<String>();
+            for (Id childrenRecordId : childRecords.keySet()) {
+                result.add(childRecords.get(childrenRecordId));
+            }
+            
+            return String.join(result, ', ');
+        }
+
+        /*******************************************************************************************************
+        * @description Get the current user Theme Context (Lightning Experience or Classic).
+        * @return Boolean.
+        ********************************************************************************************************/
+        private Boolean isLightningContext() {
+            if (LIGHTNING_CONTEXT != null) {
```

Comment:
Since this is a variable used in tests only, can we rewrite the code as following so the variable works for both tests and production code:
```
@TestVisible
private static Boolean isLightningContext {
    get {
         return isLightningContext = isLightningContext == null
             ? USER_UserService.isUserRunningLightning() || USER_UserService.isUserRunningMobile()
             : isLightningContext
    }
    set;
}
```

The variable can be set from a test code as CDL_CascadeDeleteLookups.isLightningContext = true;//or false;

Reply:
agreed with Radmila's comment.  Also, I don't like seeing variables fully capitalized, which we should be reserving for constants.